framework/include/kernels/ADTimeKernelValue.h

```diff
@@ -0,0 +1,46 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#ifndef ADTIMEKERNELVALUE_H
+#define ADTIMEKERNELVALUE_H
+
+#include "ADKernelValue.h"
+
+/**
+ * All AD time kernels should inherit from this class
```

Comment:
This is probably closer to being true but probably still not true (as you have demonstrated by your need for this PR :smile:)